interp/src/interpreter/interpret_control.rs

```diff
@@ -140,12 +101,7 @@ fn eval_while(
         return eval_while(
             w,
             continuous_assignments,
-            interpret_control_inner(
-                &w.body,
-                continuous_assignments,
-                env,
-                comp,
-            )?,
+            interpret_control(&w.body, continuous_assignments, env, comp)?,
```

Comment:
Worth taking this opportunity to turn the code to use a `loop { ...; break }` pattern instead of the recursive call since you mentioned that Rust doesn't do TCO.